lib/discord-utils.ts

```diff
@@ -0,0 +1,36 @@
+import { stripInvalidCharacters } from "./utils";
+
+// Discord truncation utility functions
+// These are pure functions that don't depend on external services
+function truncateTitle(title: string) {
+  // Discord thread title limit is 100 characters total
+  // We need to account for " - " (3 chars) + timestamp (10 chars) = 13 chars
+  // So the title absolute maximum is 87 characters
+  const titleMaxLength = 87;
+
+  if (title.length <= titleMaxLength) {
+    return title;
+  }
+
+  return `${title.substring(0, titleMaxLength - 3)}...`; // 3 char accounts for ellipsis
+}
```

Comment:
@james-a-morris this is the meaningful change. The rest is moving files around and adding tests